PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/FilterBubbles.java

```diff
@@ -0,0 +1,149 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import nl.tudelft.pl2016gr2.model.Bubble;
+import nl.tudelft.pl2016gr2.model.BubbledGraph;
+import nl.tudelft.pl2016gr2.model.GraphInterface;
+import nl.tudelft.pl2016gr2.model.IPhylogeneticTreeNode;
+import nl.tudelft.pl2016gr2.model.Node;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.test.utility.TestId;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+/**
+ * Class used to filter bubbles based on the phylogenetic tree. 
+ * @author Casper
+ *
+ */
+public class FilterBubbles {
```

Comment:
You might want to consider to more elaborately explain the functionality of this class (at the class JavaDoc or the constructor's) Reading either of these right now still doesn't give me a clear idea of what exactly it does or what the bubbles are based on exactly, even though I'm supposed to know it because we discussed it.